File: Q24/main.py
n=int(input())
arr=[]

for i in range(n):
    arr.append(int(input()))

start=mid=0
end=n-1

while(mid<=end):
    if arr[mid]==0:
        temp=arr[mid]
        arr[mid]=arr[start]
        arr[start]=temp
        start+=1
        mid+=1
    elif arr[mid]==1:
        mid+=1
    else:
        temp=arr[mid]
        arr[mid]=arr[end]
        arr[end]=temp
        end-=1

for i in range(n):
    print(arr[i],end=" ")

File: Q24/main.cpp
#include<iostream>
using namespace std;

int main(){
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }

    int start=0,mid=0,end=n-1;

    while(mid<=end){
        if(arr[mid]==0){
            int temp=arr[mid];
            arr[mid]=arr[start];
            arr[start]=temp;
            start++;
            mid++;
        }
        else if(arr[mid]==1)
            mid++;
        else{
            int temp=arr[mid];
            arr[mid]=arr[end];
            arr[end]=temp;
            end--;
        }
    }

    for(int i=0;i<n;i++){
        cout<<arr[i];
    }
    return 0;
}